src/main/scala/inox/ast/Definitions.scala

```diff
@@ -92,7 +92,7 @@ trait Definitions { self: Trees =>
     }
 
     /** Transform this [[ValDef]] into a [[Expressions.Variable Variable]] */
-    @inline def toVariable: Variable = v
+    @inline def toVariable: Variable = v.copy()
```

Comment:
Should we add `copiedFrom(v)` to cover cases where the position is expected to stay the same?

Reply:
Yes good idea, thanks